include/zephyr/drivers/spi.h

```diff
@@ -878,6 +918,117 @@ __deprecated static inline int spi_write_async(const struct device *dev,
 
 #endif /* CONFIG_SPI_ASYNC */
 
+
+#ifdef CONFIG_SPI_RTIO
+
+/**
+ * @brief Poll a SPI device with a request
+ *
+ * @param dev SPI device
+ * @param iodev_sqe Prepared submissions queue entry connected to an iodev
+ *                  defined by SPI_IODEV_DEFINE.
+ *                  Must live as long as the request is in flight.
+ *
+ * @retval 0 If successful.
+ * @retval -errno Negative errno code on failure.
+ */
+static inline enum rtio_poll_status spi_iodev_poll(struct rtio_iodev_sqe *iodev_sqe)
+{
+	const struct spi_dt_spec *dt_spec = iodev_sqe->iodev->data
+	const struct device *dev = dt_spec->bus;
+	const struct spi_driver_api *api = (const struct spi_driver_api *)bus->api;
+
+	return api->iodev_poll(dt_spec->bus, iodev_sqe);
+}
+
+/**
+ * @brief Convert a transcieve from spi_buf_sets to a sequence of RTIO ops
```

Comment:
s/transceive/transceive  

Typo seem to repeat in many place below.

Reply:
Thank you, yes I'm sure there's many other things in here that will need correction :-)